src/NewButton/counter.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import Box from '../Box'
+
+type CounterProps = {
+  count: number
+}
+
+const styles = {
+  display: 'inline-block',
+  padding: '0px 6px',
+  backgroundColor: 'neutral.muted',
+  borderRadius: '50%',
+  marginLeft: '8px',
+  marginRight: '-4px'
```

Comment:
This negative margin might create an issue if we have a count followed by an icon